src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/fetch-step/fetch-step.component.ts

```diff
@@ -0,0 +1,101 @@
+import { SetPollingSourceSection } from "./../../../../../../../dataset-block/metadata-block/components/event-details/dynamic-events/builders/set-polling-source-section.builder";
+import { RadioControlType } from "./../../form-control.source";
+import { FetchStep } from "./../../../../../../../api/kamu.graphql.interface";
+/* eslint-disable @typescript-eslint/unbound-method */
+import { BaseComponent } from "src/app/common/base.component";
+import { FormBuilder, ValidatorFn, Validators } from "@angular/forms";
+import { ControlContainer, FormGroupDirective } from "@angular/forms";
+import { ChangeDetectionStrategy, Component, OnInit } from "@angular/core";
+import { FormGroup } from "@angular/forms";
+import { fetchStepRadioControls } from "../../form-control.source";
+import {
+    JsonFormControls,
+    JsonFormData,
+    JsonFormValidators,
+} from "../../add-polling-source-form.types";
+import { fetchFormData } from "./fetch-form-data";
+
+@Component({
+    selector: "app-fetch-step",
+    templateUrl: "./fetch-step.component.html",
+    styleUrls: ["./fetch-step.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    viewProviders: [
+        { provide: ControlContainer, useExisting: FormGroupDirective },
+    ],
+})
+export class FetchStepComponent extends BaseComponent implements OnInit {
+    public parentForm: FormGroup;
+    public fetchStepRadioData: RadioControlType[] = fetchStepRadioControls;
+    public fetchFormData: JsonFormData = fetchFormData;
```

Comment:
These are not inputs, so you don't have any problem of introducing a base class and moving these there. Just use more general names for "xxxStepRadioData" and "xxxFormData"

Reply:
Ok, used more general names.